engines/optix/OptiXModel.cpp

```diff
@@ -263,6 +279,84 @@ void OptiXModel::_commitMaterials()
         material.second->commit();
 }
 
+bool OptiXModel::_commitSimulationData()
+{
+    if (!_simulationHandler)
+        return false;
+
+    if (!_setIsReadyCallback && !_animationParameters.hasIsReadyCallback())
+    {
+        auto& ap = _animationParameters;
+        ap.setIsReadyCallback(
+            [handler = _simulationHandler] { return handler->isReady(); });
+        ap.setDt(_simulationHandler->getDt(), false);
+        ap.setUnit(_simulationHandler->getUnit(), false);
+        ap.setNumFrames(_simulationHandler->getNbFrames(), false);
+        ap.markModified();
+        _setIsReadyCallback = true;
+    }
```

Comment:
This is copy pasta and it's purpose is very obscure. At least refactor it, then try to improve it.